File: Graph/python/breath_first_search.py
from collections import deque

def breath_first_search(start):
    neighbors = [] # some values...
    q = deque([start])
    seen = set()
    visited = []
    node = ''
    while len(q):
        node = q.popleft()
        if node in seen:
            continue
        visited.push(node)
        q.expand(neighbors)
        seen.add(node)
    return visited

File: Graph/js/BreathFirstSearch.js
class Node {
  constructor(value) {
    this.value = value;
    this.neighbors = [];
  }

  addNeighbor(node) {
    this.neighbors.push(node);
  }
}

function breathFirstSearch(start) {
  let q = [start];
  let a = [];
  let seen = {}
  let node;

  while (q.length > 0) {
    node = q.shift();
    if (seen[node.value]) {
      continue;
    }
    a.push(node.value);
    q = [...q, ...node.neighbors];
    seen[node.value] = true;
  }

  return a;
}